File: MultiLevel.java

class Base
{
    public int A,B;

    public Base()
    {
        System.out.println("Inside Base Constructor");
    }

    public void Fun()
    {
        System.out.println("Inside Base Fun");
    }
}

class Derived extends Base
{
    public int X,Y;

    public Derived()
    {
        System.out.println("Inside Derived Constructor");
    }

    public void Gun()
    {
        System.out.println("Inside Derived Gun");
    }  
}

class DerivedX extends Derived
{
    public int P,Q;

    public DerivedX()
    {
        System.out.println("Inside DerivedX Constructor");
    }

    public void Sun()
    {
        System.out.println("Inside DerivedX Sun");
    }  
}

class MultiLevel
{
    public static void main(String A[])
    {
        System.out.println("Inisde Main Method");
        DerivedX dobj = new DerivedX();

        dobj.Fun();
        dobj.Gun();
        dobj.Sun();
    }
}

File: Multilevel.cpp
#include<iostream>
using namespace std;

class Base
{
    public:
        int i,j;

        Base()
        {
            cout<<"Inside Base Constructor\n";
            i = 10;
            j = 20;
        }
        ~Base()
        {
            cout<<"Inside Base Destructor\n";
        }
        void Fun()
        {
            cout<<"Inside Base fun\n";
        }
};

class Derived : public Base
{
    public:
        int x,y;

        Derived()
        {
            cout<<"Inside Derived Constructor\n";
            x = 30;
            y = 40;
        }
        ~Derived()
        {
            cout<<"Inside Derived Destructor\n";
        }
        void Gun()
        {
            cout<<"Inside Derived gun\n";
        }
};

class DerivedX : public Derived
{
    public:
        int a,b;

        DerivedX()
        {
            cout<<"Inside Derived Constructor\n";
            a = 50;
            b = 60;
        }
        ~DerivedX()
        {
            cout<<"Inside Derived Destructor\n";
        }
        void Sun()
        {
            cout<<"Inside DerivedX Sun\n";
        }
};

int main()
{

    DerivedX dobj;

    cout<<"inside main function\n";

    cout<<"Size of base class "<<sizeof(Base)<<"\n";
    cout<<"Size of Derived class "<<sizeof(Derived)<<"\n";
    cout<<"Size of DerivedX class "<<sizeof(DerivedX)<<"\n";\

    cout<<dobj.i<<"\n";
    cout<<dobj.j<<"\n";
    cout<<dobj.x<<"\n";
    cout<<dobj.y<<"\n";
    cout<<dobj.a<<"\n";
    cout<<dobj.b<<"\n";

    dobj.Fun();
    dobj.Gun();
    dobj.Sun();

   // cout<<sizeof(bobj)<<"\n";
   // cout<<sizeof(dobj)<<"\n";

    return 0;
}